PPP management don't build

**Describe the bug**
samples/net/gsm_modem with CONFIG_NET_L2_PPP_MGMT=y don't build.

**To Reproduce**

1- With configuration and overlay from ee5bf09c2106065623d46069c4804f0ae7e73c0a
2- Add `CONFIG_NET_L2_PPP_MGMT=y` on configuration
3- `west build -b sam4e_xpro samples/net/gsm_modem`

**Screenshots or console output**
```
/home/gfbudke/zephyr-sdk-0.11.2/arm-zephyr-eabi/bin/../lib/gcc/arm-zephyr-eabi/9.2.0/../../../../arm-zephyr-eabi/bin/ld: zephyr/subsys/net/l2/ppp/libsubsys__net__l2__ppp.a(ppp_l2.c.obj): in function `carrier_on':
/home/gfbudke/zephyros/mainline/zephyr/subsys/net/l2/ppp/ppp_l2.c:270: undefined reference to `ppp_mgmt_raise_carrier_on_event'
/home/gfbudke/zephyr-sdk-0.11.2/arm-zephyr-eabi/bin/../lib/gcc/arm-zephyr-eabi/9.2.0/../../../../arm-zephyr-eabi/bin/ld: zephyr/subsys/net/l2/ppp/libsubsys__net__l2__ppp.a(ppp_l2.c.obj): in function `carrier_off':
/home/gfbudke/zephyros/mainline/zephyr/subsys/net/l2/ppp/ppp_l2.c:292: undefined reference to `ppp_mgmt_raise_carrier_off_event'
collect2: error: ld returned 1 exit status
```

**Environment:**
 - Debian 10
 - Zephyr SDK 0.11.2
 - mainline version
 - ee5bf09c2106065623d46069c4804f0ae7e73c0a

--- a/subsys/net/l2/ppp/CMakeLists.txt
+++ b/subsys/net/l2/ppp/CMakeLists.txt
@@ -20,6 +20,8 @@ zephyr_library_sources_ifdef(CONFIG_NET_L2_PPP
 
 zephyr_library_sources_ifdef(CONFIG_NET_STATISTICS_PPP ppp_stats.c)
 
+zephyr_library_sources_ifdef(CONFIG_NET_L2_PPP_MGMT ppp_mgmt.c)
+
 if(CONFIG_NET_IPV4)
   zephyr_library_sources_ifdef(CONFIG_NET_L2_PPP ipcp.c)
 endif()

--- a/subsys/net/l2/ppp/ppp_mgmt.c
+++ b/subsys/net/l2/ppp/ppp_mgmt.c
@@ -1,0 +1,17 @@
+/*
+ * Copyright (c) 2020 Lemonbeat GmbH
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <net/ppp.h>
+
+void ppp_mgmt_raise_carrier_on_event(struct net_if *iface)
+{
+	net_mgmt_event_notify(NET_EVENT_PPP_CARRIER_ON, iface);
+}
+
+void ppp_mgmt_raise_carrier_off_event(struct net_if *iface)
+{
+	net_mgmt_event_notify(NET_EVENT_PPP_CARRIER_OFF, iface);
+}

